health-events-analyzer/pkg/reconciler/reconciler.go

```diff
@@ -90,227 +87,172 @@ func (r *Reconciler) Start(ctx context.Context) error {
 	slog.Info("Listening for events on the channel...")
 
 	for event := range watcher.Events() {
-		startTime := time.Now()
-
-		document := event["fullDocument"].(bson.M)
+		slog.Info("Processing event", "event", event)
 
-		healthEventWithStatus := storeconnector.HealthEventWithStatus{}
-		if err := storewatcher.UnmarshalFullDocumentFromEvent(
-			event,
-			&healthEventWithStatus,
-		); err != nil {
-			slog.Error("Failed to unmarshal event", "error", err)
-			totalEventProcessingError.WithLabelValues("unamrshal_doc_error").Inc()
+		go func(ctx context.Context, event bson.M) {
```

Comment:
Why does this need to be a go routine? Can't we process events one by one as they arrive from MongoDB?